lib/main.gi

```diff
@@ -497,6 +497,7 @@ local
 		Epi := NaturalHomomorphismByNormalSubgroup( N, x[3] ) ;
     NH  := Image(Epi,N);
     KH  := Image(Epi,x[2]);
+    Reset(GlobalMersenneTwister);
```

Comment:
I will be very reluctant to reset THE `GlobalMersenneTwister`. One could of course use `InstallOtherMethodWithRandomSource` to allow to use a special random source in `Random(rs,G)`. But actually the manual says that `MinimalGeneratingSet` will be efficient in case the group is cyclic, and there is even a special method installed for this. Then the code could look like:
```
if Size(KH)=1 then
  k:=One(KH);
else
  if IsCyclic(KH) then
    k := MinimalGeneratingSet(KH)[1];
  else
    Error("KH must be a cyclic group!");
  fi;
fi;
```

Reply:
I think that we already fix this following your suggestion.
I will submit it soon